jolt-core/src/poly/commitment/dory.rs

```diff
@@ -0,0 +1,993 @@
+                // has the same memory layout as &[F]. We can transmute the slice reference.
+                let field_scalars_slice: &[F] = unsafe {
+                    std::slice::from_raw_parts(scalars.as_ptr() as *const F, scalars.len())
+                };
+                // Note: This still allocates because DensePolynomial owns its data.
+                // @TODO(markosg04) For true zero-cost, DensePolynomial needs to be refactored slightly.
+                MultilinearPolynomial::LargeScalars(DensePolynomial::new(
+                    field_scalars_slice.to_vec(),
+                ))
+            }
+            DoryMultilinearPolynomial::U8Scalars(scalars) => {
+                MultilinearPolynomial::U8Scalars(CompactPolynomial::from_coeffs(scalars.to_vec()))
+            }
+            DoryMultilinearPolynomial::U16Scalars(scalars) => {
+                MultilinearPolynomial::U16Scalars(CompactPolynomial::from_coeffs(scalars.to_vec()))
+            }
+            DoryMultilinearPolynomial::U32Scalars(scalars) => {
+                MultilinearPolynomial::U32Scalars(CompactPolynomial::from_coeffs(scalars.to_vec()))
+            }
+            DoryMultilinearPolynomial::U64Scalars(scalars) => {
+                MultilinearPolynomial::U64Scalars(CompactPolynomial::from_coeffs(scalars.to_vec()))
+            }
+            DoryMultilinearPolynomial::I64Scalars(scalars) => {
+                MultilinearPolynomial::I64Scalars(CompactPolynomial::from_coeffs(scalars.to_vec()))
+            }
+        }
+    }
+}
+
+impl<'a, F: JoltField> From<&'a MultilinearPolynomial<F>>
+    for DoryMultilinearPolynomial<'a, JoltFieldWrapper<F>>
+{
+    fn from(
+        src: &'a MultilinearPolynomial<F>,
+    ) -> DoryMultilinearPolynomial<'a, JoltFieldWrapper<F>> {
+        use DoryMultilinearPolynomial as Dory;
+
+        match src {
+            MultilinearPolynomial::LargeScalars(dense) => {
+                // SAFETY: JoltFieldWrapper is #[repr(transparent)] over F,
+                // hence always the same memory layout
+                let wrapped: &'a [JoltFieldWrapper<F>] =
+                    unsafe { std::slice::from_raw_parts(dense.Z.as_ptr().cast(), dense.Z.len()) };
+                Dory::LargeScalars(wrapped)
+            }
+            MultilinearPolynomial::U8Scalars(cp) => Dory::U8Scalars(cp.coeffs.as_slice()),
+            MultilinearPolynomial::U16Scalars(cp) => Dory::U16Scalars(cp.coeffs.as_slice()),
+            MultilinearPolynomial::U32Scalars(cp) => Dory::U32Scalars(cp.coeffs.as_slice()),
+            MultilinearPolynomial::U64Scalars(cp) => Dory::U64Scalars(cp.coeffs.as_slice()),
+            MultilinearPolynomial::I64Scalars(cp) => Dory::I64Scalars(cp.coeffs.as_slice()),
+        }
+    }
+}
+
+// ===== Transcript Adapter =====
+
+/// NewType wrapper for Jolt transcript to Dory transcript
+#[derive(Default)]
+pub struct JoltToDoryTranscriptRef<'a, F: JoltField, T: Transcript> {
+    transcript: Option<&'a mut T>,
```

Comment:
Why is this an `Option`? Seems like it is only initialized with `Some` in `new` method.